torch/nn/modules/activation.py

```diff
@@ -741,19 +744,36 @@ def _reset_parameters(self):
     @weak_script_method
     def forward(self, query, key, value, key_padding_mask=None, incremental_state=None,
                 need_weights=True, static_kv=False, attn_mask=None):
-        """
-        Inputs of forward function
-            query: [target length, batch size, embed dim]
-            key: [sequence length, batch size, embed dim]
-            value: [sequence length, batch size, embed dim]
-            key_padding_mask: if True, mask padding based on batch size
-            incremental_state: if provided, previous time steps are cashed
-            need_weights: output attn_output_weights
-            static_kv: key and value are static
-
-        Outputs of forward function
-            attn_output: [target length, batch size, embed dim]
-            attn_output_weights: [batch size, target length, sequence length]
+        r"""
+    Args:
+        query, key, value: map a query and a set of key-value pairs to an output. 
+            See "Attention Is All You Need" for more details. 
+        key_padding_mask: if provided, padding elements can be excluded from by 
+            passing a binary ByteTensor.
+        incremental_state: if provided, previous time steps are cached.
+        need_weights: output attn_output_weights.
+        static_kv: key and value are static.
+        attn_mask: mask that prevents attention from certain positions.
+
+    Shape:
+        - Inputs:
+
+        - query: :math:`(T, B, E)` where T is target length, B is batch size, E is 
```

Comment:
```suggestion
        - query: :math:`(L, N, E)` where T is target length, B is batch size, E is 
```

N is the standard notation in PyTorch documentation for the batch size see https://pytorch.org/docs/stable/nn.html#recurrent-layers. Also, I would use L for the  sequence length for query as word target is somewhat ambiguous in this context since it is not necessarily clear why you are referring to the query as a target.

Reply:
For self-attn, query, key, and value are same. However, this is not necessarily true for multi-head attention. In a sequence to sequence transform problem, the query and key may have different length. For example, the query fed to the decoder of a transformer may have different length from the query to the encoder.

Reply:
This change should also be applied to the shapes listed below.

Reply:
I am not suggesting that you force them to have the same name and integer value. I just think target and T are ambiguous choices for this name. Instead you can use L for the sequence length of the query and S for the sequence length of the value and key (similar to the choices made in the documentation nn.RNN)